### PR TITLE
feat: Generic constrs, window constrs, from hbjson cli 

### PR DESCRIPTION
**Pull request by TfedUD**
😀

### CODE REVIEW

File: dragonfly_doe2/doe/utils.py [hunk 1]
```diff
@@ -5,36 +5,49 @@ def short_name(name, max_length):
...
+        w_par = room_2d.window_parameters[:len(floor_geo.boundary)]
+    else:
+        bcs = room_2d.boundary_conditions
+        w_par = room_2d.window_parameters
+    bcs = bcs[pt_i:] + bcs[:pt_i]
+    w_par = w_par[pt_i:] + w_par[:pt_i]
+    return (verts, bcs, w_par, w_const_name)
+    # is just sticking some data needed down the line
+    # in a tuple, piggybacking on the data that is already sent
+    # to the target class: lazy? good? bad?
```
Comment (mostaphaRoudsari) [inline dragonfly_doe2/doe/utils.py:53]: I think if it is working, then it is fine.

File: dragonfly_doe2/doe/glass_types.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    """Helper function to convert values from one unit to another."""
+    for key in lbt_units:
+        if from_ in lbt_units[key]:
+            base_type = lbt_td[key]()
+            break
+    else:
+        raise ValueError(f'Invalid type: {from_}')
+
+    value = base_type.to_unit(value, to_, from_)
+    return round(value[0], 3)
```
Comment (mostaphaRoudsari) [inline dragonfly_doe2/doe/glass_types.py:25]: Don't we have this already in a different module for constructions? Can we reuse that?

File: assets/reference_dfm/complex_model_with_bcs.dfjson [hunk 3]
```diff
@@ -952,7 +952,7 @@
                 }, 
                 {
                     "type": "WindowConstructionAbridged", 
-                    "identifier": "U 0.44 SHGC 0.26 Dbl Ref-B-H Clr 6mm/13mm Air", 
+                    "identifier": "smallernamewindowconst",
```
[the remark this answers is not in the document]
Comment (TfedUD) [reply assets/reference_dfm/complex_model_with_bcs.dfjson:955]: I'd done that during some exploration: I forgot to set it back!

File: dragonfly_doe2/doe/glass_types.py (hunk 2 above)
Comment (TfedUD) [reply dragonfly_doe2/doe/glass_types.py]: Done! in utils.py

File: dragonfly_doe2/doe/glass_types.py [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+
+@dataclass
+class GlassType:
```
[the remark this answers is not in the document]
Comment (TfedUD) [reply dragonfly_doe2/doe/glass_types.py:15]: Thanks for teaching! 😃

**Comment by TfedUD** [comment]
@mostaphaRoudsari are we awaiting the HB==DF==>Doe2 to check good before the merge on this one?

**Comment by mostaphaRoudsari** [comment]
We can merge this in if that makes things easier on your side.

**Comment by TfedUD** [comment]
I think if anything was just more so the OCD manor in which I've been doing git stuff 😅,
makes fork named for feats to be added, completes that, makes another named for new feats to be added.

And also; that commit is the imo "Can use" milestone; and wanted to start making tooling off of the LBT main via BR+A github

**Pull request merged**